code_style.md

```diff
@@ -225,6 +225,12 @@ Unless otherwise specified, the following applies to all code:
     }
     ```
 
+37. Avoid functions whose behaviour / return type varies with different parameter types.
+    Multiple return types are fine, but the function's fundamental behaviour should not
+    change according its arguments: have two separate functions. For example,SDKConfig.get()
```

Comment:
I still think it's a bit weird that we say "Avoid functions whose ... return type varies" on one line and then "Multiple return types are fine" on the next line.

Maybe something like:

```suggestion
37. Avoid functions whose fundamental behaviour varies with different parameter types.
    Multiple return types are fine, but if the function's behaviour is going to change significantly,
    have two separate functions. For example, `SDKConfig.get()`
```

YMMV though. The whole thing is necessarily a bit woolly and no phrasing is going to be perfect.

Reply:
Yeah, probably better: applied manually